static/js/stream_create.js

```diff
@@ -466,4 +383,33 @@ exports.set_up_handlers = function () {
     });
 };
 
+function setup_new_sub_pills() {
+    const pill_container = $("#stream-creation .add_subscribers_container .pill-container");
+
+    function create_item_from_text(text, current_items) {
+        const item = stream_pill.create_item_from_stream_name(text, current_items);
+        if (item) {
+            return item;
+        }
+        return user_pill.create_item_from_email(text, current_items);
+    }
+
+    function get_text_from_item(item) {
+        const text = stream_pill.get_stream_name_from_item(item);
+        if (text) {
+            return text;
+        }
+        return user_pill.get_email_from_item(item);
+    }
+
+    exports.pill_widget = input_pill.create({
+        container: pill_container,
+        create_item_from_text,
+        get_text_from_item,
+    });
+
+    const opts = {stream: true};
+    pill_typeahead.set_up(pill_container.find(".input"), exports.pill_widget, opts);
```

Comment:
We could deduplicate this function by having `pill_container` and `opts` as arguments and return the `pill_widget` and with the code in `stream_edit.show_subscription_settings`.

It might be a problem to include the `opts.source` in the function, but I guess it would be okay to have the 1 liner typeahead set up below the caller code and not included in this function?

Reply:
I see this as a lot of refactoring for very short gains. I am not sure if this is a good idea.